### ISSUE TITLE
osmd.cursor.next() skips first VerticalGraphicalStaffEntryContainer (VGSEC) of score

### ISSUE THREAD

**Issue by michaelcchu** [opensheetmusicdisplay/opensheetmusicdisplay#1304]
*Note: in this post, we will abbreviate [VerticalGraphicalStaffEntryContainer](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/wiki/Music-Sheet-Object-Model#osmd-1) as VGSEC.

Steps to reproduce:

1. Open [demo](https://opensheetmusicdisplay.github.io/demo/).
2. `osmd.cursor.NotesUnderCursor()` returns `(7) [e.Note, e.Note, e.Note, e.Note, e.Note, e.Note, e.Note]`, as expected (the cursor is at the first VGSEC of the score).
3. Click "Back".
4. `osmd.cursor.NotesUnderCursor()` returns `[]`, as expected (the cursor is at the position *before* the first VGSEC of the score).
5. Click "Next".
6. `osmd.cursor.NotesUnderCursor()` returns `(3) [e.Note, e.Note, e.Note]` (the cursor is at the second VGSEC of the score). This is **unexpected**. The cursor started at the first VGSEC, moved to the position before the first VGSEC, and now should have returned to the first VGSEC. But instead, the cursor is at the second VGSEC.

In summary, we expected

`(7) [e.Note, e.Note, e.Note, e.Note, e.Note, e.Note, e.Note]` (first VGSEC),
`[]` (before first VGSEC),
`(7) [e.Note, e.Note, e.Note, e.Note, e.Note, e.Note, e.Note]` (first VGSEC),

but instead, we got

`(7) [e.Note, e.Note, e.Note, e.Note, e.Note, e.Note, e.Note]` (first VGSEC),
`[]` (before first VGSEC),
`(3) [e.Note, e.Note, e.Note]` (second VGSEC).

**Comment by sschmidTU** [comment]
Fixed.

I think the exact behavior here is a matter of preference, technically there is no value currently in going beyond the beginning of the score, and visually the cursor position also doesn't change, which could confuse users who e.g. hold down the left arrow in the demo (`previous()`) to go to the beginning of the score, overshoot it, and then see no reaction on using the right arrow (`next()`).

I also added `osmd.cursor.Iterator.currentTimeStamp` being -1 (RealValue) when going beyond the beginning of the score, analogous to it being 99999 currently when going after the end.
As I said, the exact behavior here is a matter of preference, but at least it's more consistent now.
